cdap-ui/app/cdap/components/AbstractWidget/HierarchyWidget/HierarchyTreeNode.tsx

```diff
@@ -0,0 +1,366 @@
+  const allowedTypes: string[] = widgetProps.allowedTypes || [];
+  const fieldValues = getFields(inputSchema, allowedTypes);
+  const dropdownData = inputSchemaToFlat(fieldValues);
+
+  let mrgL = 0;
+  if (hasChild && childVisible) {
+    mrgL += 6;
+  }
+
+  const fillMultiSelectOptions = (e) => {
+    setMultiSelectOptions(e);
+  };
+
+  const handleMultiSelectClose = () => {
+    const newRecords = [...records];
+    multiSelectOptions.forEach((item) => {
+      let childrenOfSelected = [];
+
+      const uniqueId = uuidV4();
+
+      if (item.children) {
+        childrenOfSelected = handleChildren(childrenOfSelected, item.children, uniqueId, 0);
+      }
+      const childToBeAdded = {
+        id: uniqueId,
+        path: item.path,
+        parentId: node.id,
+        name: item.name,
+        children: [],
+        type: item.type,
+      };
+      setShowMultiSelect(false);
+      setChildVisiblity(true);
+      newRecords.push(childToBeAdded);
+      if (childrenOfSelected) {
+        childrenOfSelected.forEach((child) => {
+          newRecords.push(child);
+        });
+      }
+    });
+    setJsonData(toJsonHandler(newRecords));
+    setRecords(newRecords);
+    setMultiSelectOptions([]);
+  };
+
+  const nameChangeHandler = (e) => {
+    const newRecords = [...records];
+    newRecords.forEach((record) => {
+      if (record.id === node.id) {
+        record.name = e.target.value;
+      }
+    });
+    setJsonData(toJsonHandler(newRecords));
+    setRecords(newRecords);
+  };
+
+  const handleAddNewRecordToParent = () => {
+    const newRecords = [...records];
+    newRecords.push({
+      id: uuidV4(),
+      parentId: node.id,
+      type: 'record',
+      children: [],
+      name: '',
+    });
+    setJsonData(toJsonHandler(newRecords));
+    setRecords(newRecords);
+    setChildVisiblity(true);
+  };
+
+  const recur = (...values) => ({ recur, values });
+
+  const loop = (f) => {
+    let a = f();
+    while (a && a.recur === recur) {
+      a = f(...a.values);
+    }
+    return a;
+  };
+
+  const removeFamily = (id = 0, nodes = []) =>
+    loop(([node, ...more] = nodes, s = new Set([id]), r = []) =>
+      node === undefined
+        ? r
+        : s.has(node.id) || s.has(node.parentId)
+        ? recur([...r, ...more], s.add(node.id), [])
+        : recur(more, s, [...r, node])
+    );
+
+  const handleDeleteElement = () => {
+    const newRecords = [...records];
+    const elements = removeFamily(node.id, newRecords);
+    setJsonData(toJsonHandler(elements));
+    setRecords(elements);
+  };
+
+  const ArrowIcon = (
+    <ButtonIcon
+      onClick={() => {
+        setChildVisiblity((value) => !value);
+        setShowMultiSelect(false);
+      }}
+    >
+      {childVisible ? <KeyboardArrowDownIcon /> : <KeyboardArrowRightIcon />}
+    </ButtonIcon>
+  );
+
+  const renderChildren = (option, selected) => {
+    return (
+      <div style={{ marginLeft: `${option.margin}px` }}>
```

Comment:
No inline-styles please. We can use `makeStyles` to pass props to css-in-js which can generate new styles dynamically. Refer to [this doc](https://material-ui.com/styles/api/#examples-3)